boards: atmel: samx7x add hsmci peripheral

tuna-f1sh opened the pull request:
Adds samx7x hsmci peripheral according to datasheet sections 10-1 and 14.1: https://ww1.microchip.com/downloads/aemDocuments/documents/MCU32/ProductDocuments/DataSheets/SAM-E70-S70-V70-V71-Family-Data-Sheet-DS60001527.pdf

Updates sam_e70_xplained overlay to include pin mapping to the SD card connected to the hsmci and adds sdhc0 alias.

Relevant https://github.com/zephyrproject-rtos/zephyr/issues/18614

tuna-f1sh committed:
boards: atmel: samx7x add hsmci peripheral

Adds samx7x hsmci peripheral according to datasheet sections 10-1 and 14.1: https://ww1.microchip.com/downloads/aemDocuments/documents/MCU32/ProductDocuments/DataSheets/SAM-E70-S70-V70-V71-Family-Data-Sheet-DS60001527.pdf

Updates sam_e70_xplained overlay to include pin mapping to the SD card
connected to the hsmci and adds sdhc0 alias.

Signed-off-by: John Whittington <git@jbrengineering.co.uk>

nandojve commented on boards/atmel/sam/sam_e70_xplained/sam_e70_xplained-pinctrl.dtsi:
```diff
@@ -182,4 +182,14 @@
 				 <PD17B_USART2_SCK>;
 		};
 	};
+	hsmci_default: hsmci_default {
```
1 line space between nodes

nandojve commented on boards/atmel/sam/sam_e70_xplained/sam_e70_xplained_same70q21b.yaml:
```diff
@@ -19,6 +19,7 @@ supported:
   - pwm
   - netif:eth
   - spi
+  - sdhc
```
sdhc should be before spi, alpha order.

nandojve commented on boards/atmel/sam/sam_e70_xplained/sam_e70_xplained-common.dtsi:
```diff
@@ -238,3 +239,16 @@ zephyr_udc0: &usbhs {
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&hsmci_default>;
+	cd-gpios = <&pioc 16 GPIO_ACTIVE_LOW>;
```
please, consider to add this to `samv71_xult` board too. All configs are the same except the `cd-gpio` which should be
`	cd-gpios = <&piod 18 GPIO_ACTIVE_LOW>;`

nandojve commented:
BTW, how are you testing this ?

tuna-f1sh committed:
boards: sam: sam_v71_xult add hsmci for sdhc

Based on 4.3.6 https://ww1.microchip.com/downloads/en/devicedoc/atmel-42408-samv71-xplained-ultra_user-guide.pdf

Signed-off-by: John Whittington <git@jbrengineering.co.uk>

tuna-f1sh commented:
Good question. I have a SAMV71 Xplained Ultra I've been building for so same family. I actually noticed I didn't change the board for that so have added a commit.

In terms of testing I tested basic build and disk access with the `sam_v71_xult/samv71q21` which works. The fs_sample fails however mounting. Reading through https://github.com/zephyrproject-rtos/zephyr/pull/58233 it looks like this is known so perhaps there is more work to this PR to get mounting working on samx7x?

```
*** Booting Zephyr OS build v4.1.0-2-gf3c3d5c8e066 ***
[00:00:01.027,000] <inf> main: Block count 62333952
Sector size 512
Memory Size(MB) 30436
[00:00:13.722,000] <err> fs: fs mount error (-19)
Error mounting disk.
[00:00:13.722,000] <err> fs: fs not mounted (mp == 0x20400000)
```

nandojve reviewed (approved):
Hi @tuna-f1sh ,

> In terms of testing I tested basic build and disk access with the sam_v71_xult/samv71q21 which works.

I think for the purpose of this PR the above is fine!
Thank you for bring the samv_71_xult too : )

Enjoy!

tuna-f1sh commented:
Cool thanks. I agree and the mount issue might actually be my hardware since basic disk io seems to work.

The SD holder part is obsolete so I have a janky setup using an adaptor and wires 🫣. The wires could be creating issues at the hsmci speed.

![image](https://github.com/user-attachments/assets/eed77c8f-bfa2-4c35-8ea5-56decaa1cf2b)
